### ISSUE TITLE
Allow everything should be removed from all levels. 

### ISSUE THREAD

**Issue by SamMousa** [WorldHealthOrganization/herams-backend#294]
> We prefer to manage permissions individually.

**Comment by SamMousa** [comment]
As exa

> We prefer to manage permissions individually.

**Comment by SamMousa** [comment]
As I have apparently failed to explain several times.
Permissions that are not granted are not shown to most users.
Only the global admins are shown the full list of permissions.

As requested I have removed the permission from the pages that allow sharing; I firmly believe that this is the wrong choice. If you cannot teach the 5? global admins on the HeRAMS team this then they should not be global admins.
These admins have permissions to delete most, if not all, stuff from the platform. Randomly denying them this one permission suggests they do not know what they are doing and if they do not then they should not be global admins in the first place.

To aid with this I have added the global `create project` permission, this way you could allow them to create projects without making them global admins.

**Comment by SamuelPetragallo** [comment]
> As I have apparently failed to explain several times.
> Permissions that are not granted are not shown to most users.
> Only the global admins are shown the full list of permissions.
> 
> As requested I have removed the permission from the pages that allow sharing; I firmly believe that this is the wrong choice. If you cannot teach the 5? global admins on the HeRAMS team this then they should not be global admins.
> These admins have permissions to delete most, if not all, stuff from the platform. Randomly denying them this one permission suggests they do not know what they are doing and if they do not then they should not be global admins in the first place.
> 
> To aid with this I have added the global `create project` permission, this way you could allow them to create projects without making them global admins.

It was clear to no-one what was granted through that "allow everything" permission, myself included. If it were just a "select all of the above" tickbox then it wouldn't be an issue, but it would also probably no longer be a need. Additionally, we need to start thinking about potential "regional admin" roles that will share the same responsibilities as global admin, but limited to certain geographical areas, meaning a potential increase in the number of users on this and headaches for anything that's unclear.

Note as mentioned that removing this also implies spelling out all individual permissions (as described in other correspondance). This may have already been done (sorry no time to check) in which case please ignore ;)

Thanks!

**Comment by SamMousa** [comment]
>  but it would also probably no longer be a need.

It will be; but we'll revert the change when we run into it...

**Comment by luna404** [comment]
Based on your explanation during the meeting yesterday I'm reopening this discussion. 

As you explained the `allow everything` position might be useful for debugging purposes in case we run into an issue with the existing permissions. Also it would only be visible to `global admins`. 

Can `global admins` share the right to `allow everything` to other users? If not, I would vote for keeping the `allow everything` option but renaming it into something like `debug mode`.

**Comment by SamMousa** [comment]
> Can global admins share the right to allow everything to other users?
Yes they can.

> If not, I would vote for keeping the allow everything option but renaming it into something like debug mode. 
This is a terrible idea; try to educate your global admins instead of renaming stuff to (supposedly) make it idiot-proof. Global admins really should not be idiots. This is like explaining a **paint brush** as a tool to paint stuff red just because the first color you thought of is red.

**Comment by SamuelPetragallo** [comment]
Well this is really just a debug mode @SamMousa 

I think it s a good idea to keep this permission as such so that we don't end up having to fix permissions in a hurry while people are working in the field, I had also not seen it like that. But then I'd want this to be named for what it is, so that it is also used knowingly

**Comment by SamMousa** [comment]
It is not used knowingly by randomly renaming it to what it is not.

It really is not a debug mode, even though it can be used for that.

**Comment by SamuelPetragallo** [comment]
Agreed, but I'd want to be used for that only, as it is also not an "allow all of the above" but rather "allow everything, incl. stuff you may not be aware of" which will create issues and potentially tensions on the ground that we don't want to have to deal with.

The less we need to train people about things the better our daily lives ;)... we already have enough (silly) requests to deal with or mistakes to recover from (and we find that training doesn't always pay off unfortunately)

**Comment by SamMousa** [comment]
> we already have enough (silly) requests to deal with or mistakes to recover from

I know how this feels.

> Agreed, but I'd want to be used for that only, as it is also not an "allow all of the above" but rather "allow everything, incl. stuff you may not be aware of" which will create issues and potentially tensions on the ground that we don't want to have to deal with.

Renaming it will not fix this; if you really want to revisit this we need to take the time. I already indicated that I did not agree with this decision, that it has not been thought through. Now just a few days later we revisit it and jump to another non-solution.

**Comment by SamuelPetragallo** [comment]
@SamMousa I have revisited this based on the assumption that there may be bugs in the permission system that we'd have to deal with while deploying, which I had assumed would not be the case following testing/release. If you're confident that we'll have no bugs on permissions or that they can be fixed in timeframes required during field implementation (i.e. almost instantly) then I'm still on my initial decision to remove it.

If not, I'm happy to keep it as a safety measure to avoid us being stuck with issues on the ground that require 2 weeks of debugging.

I know we have very different perspectives on this, but eventually we are the ones dealing with end users on a daily basis and unfortunately, calling them stupid really wouldn't help get the project going. We do try to teach and increase end users capacity as much as we can, but this goes way beyond what we can achieve as individuals. We've just received a request from Yemen with change requests such as: new workflows of the plaftorm, changing the background color of the forms, or re-importing data, because despite insistent (bold/red) messages of caution they have kept updating data on Prime after we had migrated it to Prime... If you're up for a mission to train folks in Sana'a I'm all for it ;)

Generally speaking I'm happy to discuss views and want to keep that. I've found over the years that this as always benefited the work in the end, but once we've gone around a topic and a decision has been made we need to move on. And yes I may also get it wrong at times unfortunately, but doing my best ;)

**Comment by SamMousa** [comment]
>  If you're confident that we'll have no bugs on permissions or that they can be fixed in timeframes required during field implementation (i.e. almost instantly) then I'm still on my initial decision to remove it.

I'm not even a little confident that we've covered most things in testing. I'm also not confident that we've actually implemented everything that's needed because I feel I have no idea what's needed.
Lastly I do not want to do rushed releases as they are more likely to break things than to fix things.

> I know we have very different perspectives on this, but eventually we are the ones dealing with end users on a daily basis and unfortunately, calling them stupid really wouldn't help get the project going.

I have not talked about end users at all, all of the discussed issues are not issues for end users. They are issues for global admins only, as I've repeatedly said. To emphasize: I am not calling them stupid, you are implying that they are. I think anyone who has the global admin role should be able to learn how the system works; again these are just 5 people.

> Generally speaking I'm happy to discuss views and want to keep that. I've found over the years that this as always benefited the work in the end, but once we've gone around a topic and a decision has been made we need to move on. 

I'm not the one starting these discussions though. I just noted in advance that a choice is not correct but at some point I give up since it is, as you said, not me who faces the consequences of these bad choices. Then 2 days later it turns out there was a reason why I said something was the wrong choice and we need to revisit it.

>  And yes I may also get it wrong at times unfortunately, but doing my best ;)

When I explicitly say something will not lead to the desired results from a platform / technical perspective and then we go in the wrong direction anyway it is quite frustrating. Especially since 3? days before a possible release issues are coming up that I've explicitly warned about.

I'm not in a rush to release and also don't have a problem with releasing something like you want it to be released even though that will not help you in country. So feel free to push through, but realize that this will lead to a deployment where issues will arise that will take another (long) development cycle before they can be fixed.
